pkg/linstorhelper/client.go

```diff
@@ -0,0 +1,113 @@
+package linstorhelper
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	lapi "github.com/LINBIT/golinstor/client"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Client is a LINSTOR client with convenience functions.
+type Client struct {
+	lapi.Client
+}
+
+// NewClientForCluster returns a LINSTOR client for a LINSTOR Controller managed by the operator.
+func NewClientForCluster(ctx context.Context, cl client.Client, clusterName string, options ...lapi.Option) (*Client, error) {
+	services := corev1.ServiceList{}
+	err := cl.List(ctx, &services, client.MatchingLabels{
+		"app.kubernetes.io/instance":  clusterName,
+		"app.kubernetes.io/component": "linstor-controller",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(services.Items) != 1 {
+		return nil, nil
+	}
+
+	s := services.Items[0]
+	port := 3370
+	for _, p := range s.Spec.Ports {
+		if p.Name == "api" {
+			port = int(p.Port)
+		}
+	}
+
+	u := &url.URL{
+		Scheme: "http",
+		Host:   fmt.Sprintf("%s.%s.svc:%d", s.Name, s.Namespace, port),
+	}
+
+	c, err := lapi.NewClient(append(options, lapi.BaseURL(u))...)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{*c}, nil
+}
+
+// EnsureNode ensures a node in LINSTOR matches the given node object.
+func (c *Client) EnsureNode(ctx context.Context, node lapi.Node) (*lapi.Node, error) {
```

Comment:
"Ensure" means to me just "make sure that we have one" and not that we perform updates.

Reply:
It would be an `Upsert`, but I don't really like that word. In the older helpers it was `GetOrCreateNode`, which was IMO was just wrong...

`CreateOrUpdateNode`?

Reply:
> `CreateOrUpdateNode`?

:+1: